### DISCUSSION TITLE
Font scrambled randomly

### DISCUSSION THREAD

**Discussion by Jonesie** [fabricjs/fabric.js#9740] [General]
We have a couple of users experiencing a random issue with rubbish text displaying instead of the real font characters.    They use standard system fonts or google fonts.

![image](https://github.com/fabricjs/fabric.js/assets/3420249/f558e1d9-8fb5-4095-8dc1-9f1feb36f79f)

I cant get it to fail locally so just asking here for any clues.  Im not sure if it's a fabric issue or not.  The rest of the page is ok and windows is ok   Users are both on chrome.  Apparently they need to reboot to fix it although they may not have restarted the browser correctly.

Any ideas welcome. 

Thanks

**Comment by asturur** [comment]
More than scrambled ti seems they are picking the wrong font glyph id.
What does the actual text says?
Which is the font family used on this particular example and from what computer?

**Reply by Jonesie** [reply]
This label is using Arial.
![image](https://github.com/fabricjs/fabric.js/assets/3420249/49609e21-a953-48c7-8cb2-c62aaa0830a3)

**Reply by asturur** [reply]
Te only way i can see a font go entirely wrong in this way is if you didn't use Arial as a family name but something else and you loaded a webfont that doesm't have the english alphabet but just cyrillic or chinese ( an optimized web font, possible with google fonts ).

So is important to know the context in which the broken one was renderered ( font family name and code used to load the font ).
We are diving then in specific implementation details.

Do you have details on the machine/user/language where this is NOT working?

How are you printing the barcode, are those 3of9, interleaved 2 of 5 or something else?

**Reply by Jonesie** [reply]
They are both windows users, running chrome, English only.  

I guess there could be something in the data definition of the design that has messed up the font selection but then I would expect it to happen every time and a reboot wouldnt fix it.

Youve given me a few questions to ask the customers so I have a starting point now.  Thank you!

Not totally sure what that barcode is, we support about 20 or so.  It's not a font -- we just draw the bars/qr code with fabric and to printers/pdf.

Cheers
